library/Icingadb/Web/Adapter/QueryAdapter.php

```diff
@@ -0,0 +1,85 @@
+<?php
+/* Icinga Web 2 | (c) 2013 Icinga Development Team | GPLv2+ */
+namespace Icinga\Module\Icingadb\Web\Adapter;
+
+use ipl\Stdlib\Contract\Paginatable;
+use Zend_Paginator_Adapter_Interface;
+
+class QueryAdapter implements Zend_Paginator_Adapter_Interface
```

Comment:
Why is that necessary? Is it because of the `joystickPagination` view script? If so, why don't you just copy this view script and adjust it as well to make it work with a `ipl\Stdlib\Contract\Paginatable`?

Reply:
Yes, it is because of the `joystickPagination`. Ok, I will do as you suggested.

Reply:
Then please make it so that `PivotTable` doesn't require this. We have `ipl\Stdlib\Contract\Paginatable` as successor for that.

We don't want to use any Zend_* classes/interfaces in this module.

Reply:
Ok